ui/src/app/views/breadcrumb/component.ts

```diff
@@ -86,13 +86,23 @@ export class BreadcrumbComponent implements OnInit, OnDestroy {
         label = child.snapshot.data['cohort'].name;
       }
       if (label === 'Param: Concept Sets Name') {
-        label = child.snapshot.data['workspace'].name + ' Concepts';
+        label = child.snapshot.data['workspace'].name;
         if (breadcrumbs.length > 2) {
           breadcrumbs = breadcrumbs.filter(b => !b.url.endsWith('/concepts'));
         }
       }
       if (label === 'Param: Concept Set Name') {
         label = child.snapshot.data['conceptSet'].name;
+        // For the most part, we don't want to append the current label to the breadcrumbs
+        //      since we want the label to stop at the parent.
+        // In this case, we want the breadcrumb header to be the Concept Set Name
+        const conceptSetBreadcrumb = BreadcrumbComponent.makeBreadcrumb(label, url, child);
+        breadcrumbs.push(conceptSetBreadcrumb);
+      }
+      // Return if  child is a leaf, or if child's child has same breadcrumb as its parent
```

Comment:
Per slack discussion, I think something like an `intermediateBreadcrumb` parameter might make sense to address these cases. Reading more, I'm not actually sure I follow the logic here; why would the child have the same breadcrumb as the parent?